wordpress-mobile/WordPress-Android
# Dashboard Card: Show Activity Card

Parent #18210 

This PR adds support for showing the Activity Card on the dashboard. It includes the following:
- Created `ActivityCardBuilder` which will check if the activity card should be built and if so, build the card
- Created `ActivityCardBuilderParams` to pass along to the builder
- Build/show card logic added to `CardsBuilder`, `MySiteViewModel`, `MySiteCardAndItemBuilderParams`
- Built out `ActivityItem` to include display date, icon, and icon background color.
- Updated the activity card items in `ActivityCardViewHolder`
- Added the remaining fields to `ActivityItemViewHolder`
- Update `my_site_activity_card_item` to separate subtitle into three individual fields to accommodate truncating to a single line of text
- Added/updated string, styles, and dimen resources
- Fix existing tests and add tests to `cards/CardsBuilderTest` and `dashboard/CardsBuilderTest`
- Added `ActivityCardBuilderTest`

Light | Dark
-- | --
<img width="250" height="500" alt="Alt desc" src="https://user-images.githubusercontent.com/506707/232816839-6110a442-47d9-4a1e-996f-a97ad179e47c.png"> | <img width="250" height="500" alt="Alt desc" src="https://user-images.githubusercontent.com/506707/232816864-07c75091-3606-4608-855d-da23d416074a.png">

Notes
- This PR is the second in a series of five; and as such please consider this PR an interim step. My intent was to keep the number of commits to a reasonable number for each PR to make it easier to review.
- The activity card has undergone some design changes that are included in the next PR. The major change between this and the next PR is the order of the activity list items within the card. 
- There are numerous refactors to `my_site_activity_card_item.xml` that occur in the next branch such as: margin/padding consolidation, renaming of fields, removal of deprecated attributes, and the removal of the internal constraint layout. Please ignore these in this PR
- Navigation upon card click and activity card item click are not included in this PR

cc: @startuptester 

**To test:**
Make sure the activity card is not shown when the feature is disabled
1. Install the Jetpack app
2. Login with a wp account
3. Select a site in which you can access the activity log (the site must be a non-P2 Jetpack-connected site that supports the REST API, and the user must be an admin)
4. Navigate to App Setting > Debug Settings 
5. Disable `dashboard_card_activity_log` and restart the app
6. Navigate to Home
7. ✅ Verify that the activity card is not shown
------------------
Make sure the activity card is shown when the feature is enabled and user has access to activity log
1. Install the Jetpack app
2. Login with a wp account
3. Select a site in which you can access the activity log (manage content/admin)
4. Navigate to App Setting > Debug Settings 
5. Enable `dashboard_card_activity_log` and restart the app
6. Navigate to Home
7. ✅ Verify that the activity card is shown
9. ✅ Verify the card matches the design in this PR
----------------
Make sure the activity card is not shown when the feature is enabled and user does not have access to activity log
1. Install the Jetpack app
2. Login with a wp account
3. Select a site in which you can't access the activity log (the site can be a P2 Jetpack-connected site that doesn't support the REST API, or the user must not be an admin)
4. Navigate to App Setting > Debug Settings 
5. Enable `dashboard_card_activity_log` and restart the app
6. Navigate to Home
7. ✅ Verify that the activity card is not shown

## Regression Notes
1. Potential unintended areas of impact
The activity card is not shown on the dashboard

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit testing

3. What automated tests I added (or what prevented me from doing so)
Updated `cards/CardsBuilderTest` and `dashboard/CardsBuilderTest`
Added `ActivityCardBuilderTest`

PR submission checklist:
